File: CpuInEvent.java
/**
 * Created by veraxmedax on 03/06/2017.
 */
public class CpuInEvent extends Event {
    private Cpu cpu;

    public CpuInEvent(Simulation sim, Task task, double time, Cpu cpu) {
        setSim(sim);
        setTask(task);
        setTime(time);
        setCpu(cpu);
    }

    @Override
    void process() {
        double duration = getTime() - (getTask().getArrivalTime());
        getTask().setWaitCpu(duration);
        getCpu().work(getTask(), getTime());
        CpuOutEvent cpuOutEvent = new CpuOutEvent(getSim(), getTask(), getTask().getCpuServiceStop(), getCpu());
        getSim().schedule(cpuOutEvent);
    }

    public Cpu getCpu() {
        return cpu;
    }

    public void setCpu(Cpu cpu) {
        this.cpu = cpu;
    }
}

File: CpuInEvent.cpp
//
// Created by Can Candan on 04/06/17.
//

#include "CpuInEvent.h"
#include "CpuOutEvent.h"

CpuInEvent::CpuInEvent(Simulation *pSimulation, Task *pTask, double d, Cpu *pCpu) : Event(pSimulation, pTask, d) {
    this->cpu=pCpu;
    name="CpuInEvent";
}

void CpuInEvent::process() {
    log();
    task->setWaitCpu(time - task->getArrivalTime());
    cpu->work(task, time);

    CpuOutEvent* event= new CpuOutEvent(sim, task, task->getCpuServiceStopTime(), cpu);
    sim->schedule(event);
}

